internal/service/service.go

```diff
@@ -31,112 +31,157 @@ func NewService(repository *repository.Repository) *Service {
 	}
 }
 
-func (t Service) CreatePosition(p internal.Position) error{
+func (t Service) CreatePosition(p internal.Position) error {
 	m := t.repo.GetPositions()
 	for _, value := range m {
-		if value == p {
+		if value.Salary == p.Salary && value.Name == p.Name {
 			return errors.New("create error: this position is exists")
 		}
 	}
+	p.ID = uuid.New()
 	t.repo.AddPosition(p)
 	return nil
 }
 
-func (t Service) CreateEmployee(e internal.Employee) error{
+func (t Service) CreateEmployee(e internal.Employee) error {
 	m := t.repo.GetEmployees()
 	for _, value := range m {
-		if value == e {
+		if value.LasName == e.FirstName &&
+			value.FirstName == e.FirstName &&
+			value.PositionID == e.PositionID{
```

Comment:
Logically the same person can't work on different positions at once

Reply:
+